File: contests/abc081/abc081_b.py
# B - Shift only
# https://atcoder.jp/contests/abc081/tasks/abc081_b

n = int(input())
arr = list(map(int, input().split()))
count = 0
while all(a % 2 == 0 for a in arr):
  arr = [i / 2 for i in arr]
  count += 1
  
print(count)

File: contests/abc081/abc081_b.cpp
// B - Shift only
// https://atcoder.jp/contests/abc081/tasks/abc081_b

#include <iostream>
using namespace std;

int main() {
    int n; cin >> n;
    int a[200];
    int dmin = 100;
    for (int i=0; i<n; i++) {
        cin >> a[i];
        int d = 0;
        while (a[i]%2 == 0) {
            a[i] /= 2;
            ++d;
        }
        dmin = min(dmin, d);
    }
    cout << dmin << endl;
}
